### PR TITLE
Report unexpected characters in hex literal

### LINKED ISSUE #5: Report error for invalid hexadecimal digits: 0x12abq
quick-lint-js reports an error if a decimal or binary number literal contains bogus digits:

```javascript
// hello.js:2:5: error: unexpected characters in number literal
1234abcd
```
Give hexadecimal literals the same error-reporting treatment. See parse_binary_number in lex.cpp, and tests in test-lex.cpp.

### PR DESCRIPTION
**Pull request by welenofsky**
This may resolve #5 if I understood the request.

### CODE REVIEW

File: src/lex.cpp [hunk 1]
```diff
@@ -628,18 +632,22 @@ void lexer::parse_binary_number() {
...
+        break;
       default:
         goto done_parsing_garbage;
     }
   }
-done_parsing_garbage:
+  done_parsing_garbage:
   const char8* garbage_end = input;
   if (garbage_end != garbage_begin) {
     this->error_reporter_->report(error_unexpected_characters_in_number{
```
Comment (strager) [inline src/lex.cpp:653]: Line 658: report_garbage_in_number_literal accepts the input pointer, the assigns to `this->input_` at the end. This is different from every other function in this file; functions either exclusively manipulate `this->input_`, or they accept the input pointer and return the modified input pointer. I think report_garbage_in_number_literal should do one or the other. For example:

```cpp
const char8* lexer::report_garbage_in_number_literal(const char8* input) {
```

File: src/lex.cpp [hunk 2]
```diff
@@ -628,18 +632,22 @@ void lexer::parse_binary_number() {
     input += 1;
   }
 
+  report_garbage_in_number_literal(input);
+}
+
+void lexer::report_garbage_in_number_literal(const char8* input) {
```
Comment (strager) [inline src/lex.cpp:638]: Nit: I think `check_garbage_in_number_literal` is a better name, since this function doesn't always report an error.

Comment (strager) [inline src/lex.cpp:638]: Thanks for making a function! It's much easier to read `lexer::parse_binary_number` now.

**Review by strager — APPROVED** [review]
Looks good! Let me know if you want to make changes, or if I should land this PR as-is.

File: src/lex.cpp (hunk 2 above)
Comment (welenofsky) [reply src/lex.cpp]: Updated in latest commit. I agree that is a better name.

File: src/lex.cpp (hunk 1 above)
Comment (welenofsky) [reply src/lex.cpp]: Updated in latest commit.

**Pull request merged**